migrations/lib/importer/discourse_db.rb

```diff
@@ -93,15 +95,31 @@ def column_names(table_name)
     end
 
     def query_array(sql, *params)
+      query_result(sql, *params).rows
+    end
+
+    def query_result(sql, *params)
       @connection.send_query_params(sql, params)
       @connection.set_single_row_mode
 
-      Enumerator.new do |y|
-        while (result = @connection.get_result)
-          result.stream_each_row { |row| y.yield(row) }
-          result.clear
+      first_result = @connection.get_result
+      return QueryResult.new(rows: Enumerator.new {}, column_count: 0) unless first_result
+
+      column_count = first_result.nfields
+      single_column = column_count == 1
+
+      rows_enumerator =
+        Enumerator.new do |y|
+          first_result.stream_each_row { |row| single_column ? y << row[0] : y << row }
+          first_result.clear
+
+          while (result = @connection.get_result)
+            result.stream_each_row { |row| single_column ? y << row[0] : y << row }
+            result.clear
+          end
         end
```

Comment:
While the inline `single_column` check on each streamed row seems like an additional overhead, it performs just as well as a top-level conditional with separate enumerators or a lambda-based approach with YJIT enabled . Also, `y << row` appears to be slightly faster than `y.yield(row)`